python_standards.md

```diff
@@ -0,0 +1,167 @@
+We should use 4 space indentation in our code. Ensure that your IDE is set up so that indents represent 4 spaces and not tabs. When requiring multiple lines, always align with the opening delimiter.
+```{python}
+# Bad:  
+
+foo = long_function_name(var_one, var_two,  
+    var_three, var_four)  
+
+my_list = [1, 2, 3,   
+    4, 5, 6,]  
+
+# Good:  
+
+foo = long_function_name(var_one, var_two,  
+                         var_three, var_four)  
+
+my_list = [1, 2, 3,   
+           4, 5, 6,]
+```
+#### Using Line spaces - Sparse is better than dense
+- Surround top-level function and class definitions with two blank lines.
+- Method definitions inside a class are surrounded by a single blank line.
+- Extra blank lines may be used (sparingly) to separate groups of related functions. Blank lines may be omitted between a bunch of related one-liners (e.g. a set of dummy implementations).
+- Use blank lines in functions, sparingly, to indicate logical sections.
+#### Using White spaces
+Avoid extraneous whitespace in the following situations:
+- Immediately inside parentheses, brackets or braces
+- Between a trailing comma and a following close parenthesis
+- Immediately before a comma, semicolon, or colon
+- Immediately before the open parenthesis that starts the argument list of a function call
+- Immediately before the open parenthesis that starts an indexing or slicing
+- More than one space around an assignment (or other) operator to align it with another
+  
+Some examples below:
+```{python}
+# Good:  
+spam(ham[1], {eggs: 2})  
+
+# Bad:  
+spam( ham[ 1 ], { eggs: 2 } )  
+
+# Good:  
+if x == 4: print x, y; x, y = y, x  
+
+# Bad:  
+if x == 4 : print x , y ; x , y = y , x  
+
+# Good:  
+ham[1:9], ham[1:9:3], ham[:9:3], ham[1::3], ham[1:9:]  
+ham[lower:upper], ham[lower:upper:], ham[lower::step]  
+ham[lower+offset : upper+offset]  
+ham[: upper_fn(x) : step_fn(x)], ham[:: step_fn(x)]  
+ham[lower + offset : upper + offset]  
+
+# Bad:  
+ham[lower + offset:upper + offset]  
+ham[1: 9], ham[1 :9], ham[1:9 :3]  
+ham[lower : : upper]  
+ham[ : upper]  
+
+# Good:  
+spam(1)  
+
+# Bad  
+spam (1)  
+
+# Good:  
+dct['key'] = lst[index]  
+
+# Bad:  
+dct ['key'] = lst [index]  
+
+# Good  
+def complex(real, imag=0.0):  
+    return magic(r=real, i=imag)  
+
+# No:  
+def complex(real, imag = 0.0):  
+    return magic(r = real, i = imag)
+
+```
+### Naming
+There are many opinions on the best way to name variables in python. We will generally follow the guidelines in [PEP 8 - Naming Conventions](https://peps.python.org/pep-0008/#naming-conventions)
+#### Naming conventions
+- Variable names should be no less than 3 characters and should be ddscriptive of what the variable represents
+- Use CAPITALS to define constants or globals that will be used in your entire program or module
+- Use lowercase_with_underscores for local variables, member attributes or methods, function arguments and parameters
+- Use PascalCase for classes
```

Comment:
Is this consistent with https://github.com/HHS/acf-datasurge-standards/blob/main/script.md#case-conventions?

Reply:
We should update the script.md file to include a note that global variables should be capitalized and class definitions should use pascal or camel case (at least in python). Thanks for noting this! I'll open an issue.